cordova-lib/src/cordova/restore-util.js

```diff
@@ -63,7 +64,27 @@ function installPlatformsFromConfigXML(platforms, opts) {
             // Combining arrays and checking duplicates.
             comboArray = pkgJsonPlatforms.slice();
         }
+
         engines = cfg.getEngines(projectHome);
+
+        if(fs.existsSync(pkgJsonPath)) {
```

Comment:
I don't think you need lines 70-72 because you check if pkgJson exists at line 54.